rfcs/reftest_simplification.md

```diff
@@ -0,0 +1,56 @@
+
+We also have a number of "tests" which aren't actually found in the manifest, because we have a cycle of "tests" (and therefore we have no source node among them); these are given [here](https://gist.github.com/gsnedders/ce17e7df69ee89ce471d5a553b826dd2). Note that the majority of these are tracked in [#5294](https://github.com/web-platform-tests/wpt/issues/5492) and fixed in [#15523](https://github.com/web-platform-tests/wpt/pull/15523), and the rest are broken `/infrastructure` tests that are meant to test for support of cycles.
+
+### Proposal
+
+This RFC contains a number of proposals, given below:
+
+#### Test Detection
+
+With the current logic, we need to process every changed file in the repository before running a single given path, as we need to recompute the reftest graph to determine whether that path is still a source node in the graph. This current behaviour also sometimes causes confusion from users when files they believe to be tests do not appear in the results.
+
+While purely judging whether or not something is a test from the path (presumably with [`not name_is_reference`](https://github.com/web-platform-tests/wpt/blob/6111e21ca42dea24a07df1acfbed78769a415b5b/tools/manifest/sourcefile.py#L335)) will lead to redundant tests (e.g, `test-001.html == test-002.html == test-ref.html` will now lead to two tests), this matches user expectations more clearly and makes it possible to do partial manifest updates for reftests.
+
+#### Valid Graphs
+
+It becomes much easier to explain why a reftest fails if we have less complexity in what is allowed within the graph. Given we have so few complex cases, it seems like we should be able to massively simplify what we allow.
+
+In short, the proposal is that from a test node:
+
+ * If the out degree of the test node is > 1, then each reachable node from the test node must have an out degree of 0. (i.e., if we have alternates, no reference can have a `<link rel=match>`)
+ * If the out degree of the test node is 1, then every reachable node from the test node must have an out degree of ≤ 1. (i.e., if we have conjunctions, it's a simple chain with no alternation)
+ 
+## Advantages
+
+* Simplify runner implementations,
+* Makes it simpler to explain failure to users.
+
+## Disadvantages
+
+* Unable to deal with more complex cases that seemingly never appear in the wild but have been occasionally stated that we need.
```

Comment:
Can you find those claims? Perhaps there's merit to them if we ask a bit more. (I hope not, of course.)

Reply:
I couldn't, sadly. 🙁

Pretty sure this was many, many years ago.

Reply:
Maybe omit this point if the claims can't be found

Reply:
I was hoping someone from the CSS WG would chime in and tell me why :)